Source/BoundaryConditions/WarpX_PEC.cpp

```diff
@@ -49,9 +49,15 @@ PEC::ApplyPECtoEfield (std::array<std::unique_ptr<amrex::MultiFab>, 3>& Efield,
     amrex::IntVect Ex_nodal = Efield[0]->ixType().toIntVect();
     amrex::IntVect Ey_nodal = Efield[1]->ixType().toIntVect();
     amrex::IntVect Ez_nodal = Efield[2]->ixType().toIntVect();
-    int nComp_x = Efield[0]->nComp();
-    int nComp_y = Efield[1]->nComp();
-    int nComp_z = Efield[2]->nComp();
+    // For each Efield multifab, apply PEC boundary condition to ncomponents
+    // If not split E-field, the PEC is applied to the regular Efield used in Maxwell's eq.
+    // If split_pml_field is true, then PEC is applied to the first two components of the pml multifabs
```

Comment:
In this comment I would simply write "PEC is applied to all components of the PML MultiFabs`. I mean, I think we want to apply it to all components, rather than just the first two, or am I misunderstanding something?

Reply:
the third one is the normal component. We apply PEC only to the tangential components of the E-field. The third component when divE cleaning is on, is xx (definition in PMLComponent.H). We do not apply PEC to that component.

Reply:
As discussed in an offline meeting, PEC is applied to all split fields of the tangential components